File: 2026/04-April/15-Wednesday/Solution.go
import "math"

func closestTarget(words []string, target string, startIndex int) int {
    n := len(words)
    ans := math.MaxInt32

    for i := 0; i < n; i++ {
        if words[i] == target {
            d := abs(i - startIndex)
            ans = min(ans, min(d, n-d))
        }
    }

    if ans == math.MaxInt32 {
        return -1
    }
    return ans
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: 2026/04-April/15-Wednesday/Solution.js
var closestTarget = function(words, target, startIndex) {
    let n = words.length;
    let ans = Infinity;

    for (let i = 0; i < n; i++) {
        if (words[i] === target) {
            let d = Math.abs(i - startIndex);
            ans = Math.min(ans, Math.min(d, n - d));
        }
    } 

    return ans === Infinity ? -1 : ans;
};
